java-codegen/src/main/resources/org/opensearch/client/codegen/templates/ObjectShape/Equals.mustache

```diff
@@ -0,0 +1,21 @@
+    public boolean equals(Object o) {
+        {{#extendsOtherShape}}
+            if(!super.equals(o)) {
+                return false;
+            }
+        {{/extendsOtherShape}}
+        if(this == o) return true;
+        if(this.getClass() != o.getClass()) return false;
+        {{className}} other = ({{className}})o;
+        {{#fields}}
```

Comment:
The if chains get quite long and verbose for classes with many fields. While not critical as these files aren't to be hand edited anyway, we could possibly collapse this to a `&&` chain.

Reply:
changed to a `&&` chain at [8a150f3](https://github.com/opensearch-project/opensearch-java/pull/1201/commits/8a150f31e2ea4a60f2ebdc48ecadefd6d592108f)